File: backend/src/main/java/com/rentease/entity/Property.java
package com.rentease.entity;

import jakarta.persistence.*;
import jakarta.validation.constraints.DecimalMin;
import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.NotNull;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Entity
@Table(name = "properties")
@Data
@NoArgsConstructor
@AllArgsConstructor
public class Property {
    
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    
    @NotBlank(message = "Title is required")
    @Column(nullable = false)
    private String title;
    
    @NotBlank(message = "Address is required")
    @Column(nullable = false)
    private String address;
    
    @NotNull(message = "Rent amount is required")
    @DecimalMin(value = "0.0", inclusive = false, message = "Rent amount must be greater than 0")
    @Column(nullable = false)
    private Double rentAmount;
    
    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "owner_id", nullable = false)
    private User owner;
}

File: frontend/lib/models/property.dart
class Property {
  final int? id;
  final String title;
  final String address;
  final double rentAmount;
  final int? ownerId;
  final String? ownerName;

  Property({
    this.id,
    required this.title,
    required this.address,
    required this.rentAmount,
    this.ownerId,
    this.ownerName,
  });

  factory Property.fromJson(Map<String, dynamic> json) {
    return Property(
      id: json['id'],
      title: json['title'],
      address: json['address'],
      rentAmount: (json['rentAmount'] as num).toDouble(),
      ownerId: json['ownerId'],
      ownerName: json['ownerName'],
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'title': title,
      'address': address,
      'rentAmount': rentAmount,
    };
  }
}
